samples/cpp/benchmark_app/main.cpp

```diff
@@ -485,9 +485,10 @@ int main(int argc, char* argv[]) {
                                            });
 
             topology_name = model->get_friendly_name();
-            // use batch size according to provided layout and shapes (static case)
-            if (!isDynamicNetwork) {
-                batchSize = getModelInputBatchSize(*model);
+
+            // Calculate batch size according to provided layout and shapes (static case)
+            if (!isDynamicNetwork && app_inputs_info.size()) {
+                batchSize = getBatchSize(app_inputs_info.front());
```

Comment:
Why do you get front? Maybe it is better to use maximum batch size for case where one input has batch size > 1 and others not. I'm not sure that input with batch size > 1 will be always first.

Reply:
This is code for static case, where each input has constant shape for every test. That's why we can use any test to calculate batch - I've taken first one (app_inputs_info is a vector of InputsInfo, each element contains info for every input, for particular test configuration).